src/ansys/dpf/core/property_fields_container.py

```diff
@@ -21,227 +21,63 @@
 # SOFTWARE.
 
 """
-MockPropertyFieldsContainer.
+PropertyFieldsContainer.
 
-Contains classes associated with the _MockPropertyFieldsContainer.
+Contains classes associated with the PropertyFieldsContainer.
 """
 
 from __future__ import annotations
 
-from collections.abc import Sequence
-import copy
-from typing import Dict, List, Union
+from typing import TYPE_CHECKING, Optional
 
-import numpy as np
+from ansys.dpf.core import property_field
+from ansys.dpf.core.collection import Collection
 
-import ansys.dpf.core as dpf
-from ansys.dpf.core import PropertyField
-from ansys.dpf.core.server_types import BaseServer
+if TYPE_CHECKING:
+    from ansys.dpf.core.property_fields_container import PropertyFieldsContainer
```

Comment:
@moe-ad why is this needed? Doesn't `from __future__ import annotations` already allow to reference the current class as a type?